File: exercism/go/acronym/acronym.go
package acronym

import (
	"strings"
)

// Abbreviate should have a comment documenting it.
func Abbreviate(s string) string {
	// simple solution
	acr := ""
	// FieldsFunc could also be used
	s = strings.ReplaceAll(strings.ReplaceAll(s, "_", ""), "-", " ")

	words := strings.Fields(s)

	for _, word := range words {
		// strings.Join can be used
		acr += string([]rune(word)[0])
	}

	return strings.ToUpper(acr)
	// using a regex would result in a more elegant solution
	// re := regexp.MustCompile(`[\PL]*([\pL])[\pL']*[\PL]*`)
}

File: exercism/python/acronym/acronym.py
"""
Acronym
"""

import re


def abbreviate(words):
    """
    simple acronym:
    a = words[0]
    words = words.replace("_", "").replace("- ", "")

    for index, char in enumerate(words):
        if char == ' ' or char == '-':
            a = a + words[index + 1]
    return a.upper()
    """

    # smart: get the words
    return ''.join(word[0].upper() for word in re.findall(r"[a-zA-Z']+", words))
